lib/matplotlib/axes/_axes.py

```diff
@@ -6573,7 +6573,7 @@ def hist(self, x, bins=None, range=None, density=None, weights=None,
         # basic input validation
         input_empty = np.size(x) == 0
         # Massage 'x' for processing.
-        if input_empty:
+        if input_empty and len(x) == 0:
```

Comment:
Or more simply, this entire if...else block can be deleted and replaced by `x = cbook._reshape_2D(x, 'x')` which handles empty inputs just fine.

Reply:
That'll be much cleaner. 
But it seems  `_reshape_2D` currently doesn't work with an empty list `[]`:

```python
>>> mpl.cbook._reshape_2D([], 'x')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/hershen/matplotlib/lib/matplotlib/cbook/__init__.py", line 1418, in _reshape_2D
    if X.ndim == 1 and not isinstance(X[0], collections.abc.Iterable):
IndexError: index 0 is out of bounds for axis 0 with size 0
```

Is it reasonable to modify it so that it returns `[[]]` in such a case?

Reply:
Thanks for noticing that, that's actually a regression due to https://github.com/matplotlib/matplotlib/pull/11921 that I've re-reported in #13392, which will need to get fixed.  I think best would be for this PR to wait for #13392, but I'm not going to hold it up on that.

Reply:
Well, I noticed it only because of your suggestion ;)
I don't mind waiting for #13392.

Reply:
It's fixed now.